src/progress/pct/v11/dynrun.p

```diff
@@ -0,0 +1,136 @@
+DEFINE VARIABLE zz AS INTEGER     NO-UNDO.
+DEFINE VARIABLE xx AS INTEGER     NO-UNDO.
+DEFINE VARIABLE out1 AS CHARACTER   NO-UNDO.
+DEFINE VARIABLE out2 AS CHARACTER   NO-UNDO.
+
+ASSIGN jsonParser = NEW ObjectModelParser().
+ASSIGN configJson = CAST(jsonParser:ParseFile(SESSION:PARAMETER), JsonObject).
+
+ASSIGN pctVerbose = configJson:getLogical("verbose").
+
+// DB connections + aliases
+ASSIGN dbEntries = configJson:GetJsonArray("databases").
+DO zz = 1 TO dbEntries:Length:
+  ASSIGN dbEntry = dbEntries:GetJsonObject(zz).
+  IF pctVerbose THEN MESSAGE "Connecting to '" + dbEntry:getCharacter("connect") + "'".
+  CONNECT VALUE(dbEntry:getCharacter("connect")) NO-ERROR.
+  IF ERROR-STATUS:ERROR THEN DO:
+    MESSAGE "Unable to connect to '" + dbEntry:getCharacter("connect") + "'".
+    DO i = 1 TO ERROR-STATUS:NUM-MESSAGES:
+      MESSAGE ERROR-STATUS:GET-MESSAGE(i).
+    END.
+    RUN returnValue(14).
+    QUIT.
+  END.
+  DO xx = 1 TO dbEntry:getJsonArray("aliases"):Length:
+    IF pctVerbose THEN MESSAGE SUBSTITUTE("Creating alias &1 for database #&2 &3", dbEntry:getJsonArray("aliases"):getCharacter(xx), zz, LDBNAME(zz)).
+    CREATE ALIAS VALUE(dbEntry:getJsonArray("aliases"):getCharacter(xx)) FOR DATABASE VALUE(LDBNAME(zz)) NO-ERROR.
+    IF ERROR-STATUS:ERROR THEN DO:
+      MESSAGE SUBSTITUTE("Unable to create alias '&1' for database '&2'", dbEntry:getJsonArray("aliases"):getCharacter(xx),  LDBNAME(zz) ).
+      DO i = 1 TO ERROR-STATUS:NUM-MESSAGES:
+        MESSAGE ERROR-STATUS:GET-MESSAGE(i).
+      END.
+      RUN returnValue(15).
+      QUIT.
+    END.
+  END.
+END.
+
+// PROPATH entries
+ASSIGN ppEntries = configJson:GetJsonArray("propath").
+DO zz = 1 TO ppEntries:Length:
+    ASSIGN PROPATH = ppEntries:getCharacter(zz) + "," + PROPATH.
+END.
+IF pctVerbose THEN MESSAGE "PROPATH : " + PROPATH.
+
+// Input parameters
+ASSIGN prmEntries = configJson:GetJsonArray("parameters").
+DO zz = 1 TO dbEntries:Length:
+  ASSIGN prmEntry = prmEntries:GetJsonObject(zz).
+  DO ON ERROR UNDO, LEAVE:
+    CREATE ttParams.
+    ASSIGN ttParams.key = prmEntry:getCharacter("name")
+           ttParams.val = prmEntry:getCharacter("value").
+  END.
+END.
+
+// Output parameters
+ASSIGN outprmEntries = configJson:GetJsonArray("output").
+
+// Execute procedure
+IF pctVerbose THEN MESSAGE "RUN " + configJson:getCharacter("procedure").
+RunBlock:
+DO ON QUIT UNDO, RETRY:
+  IF RETRY THEN DO:
+    MESSAGE "QUIT statement found".
+    IF noErrorOnQuit THEN i = 0. ELSE i = 66.
+    LEAVE RunBlock.
+  END.
+  IF (outprmEntries:Length EQ 0) THEN
+    RUN VALUE(configJson:getCharacter("procedure")) NO-ERROR.
+  ELSE IF (outprmEntries:Length EQ 1) THEN
+    RUN VALUE(configJson:getCharacter("procedure")) (OUTPUT out1) NO-ERROR.
+  ELSE IF (outprmEntries:Length EQ 2) THEN
+    RUN VALUE(configJson:getCharacter("procedure")) (OUTPUT out1, OUTPUT out2) NO-ERROR.
+END.
+IF ERROR-STATUS:ERROR THEN ASSIGN i = 1.
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Statement ASSIGN should be on a separate line [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://sonar.riverside-software.fr/coding_rules#rule_key=rssw-oe-main%3Aeu.rssw.antlr.proparse.checks.OneStatementPerLine)